src/curve.rs

```diff
@@ -10,81 +15,102 @@ pub struct Curve<F: Field> {
   two:   F,
 }
 
-pub struct CurvePoint<F: Field> {
-  pub curve: Curve<F>,
-  pub point: PointOrInfinity<F>,
-}
+pub trait CurveParams: 'static + Copy + Clone + fmt::Debug + Default + Eq + Ord {
+  /// Integer field element type
+  type FieldElement: Field + Neg;
+  /// Order of this elliptic curve, i.e. number of elements in the scalar field.
+  const ORDER: u32;
+  /// Coefficient `a` in the Weierstrass equation of this elliptic curve.
+  const EQUATION_A: Self::FieldElement;
+  /// Coefficient `b` in the Weierstrass equation of this elliptic curve.
+  const EQUATION_B: Self::FieldElement;
+  /// Generator of this elliptic curve.
+  const GENERATOR: (Self::FieldElement, Self::FieldElement);
```

Comment:
This will be the generator for g1, but when we get into field extensions, I guess we will also need a g2 generator? Perhaps those are represented elsewhere. 

Also curious how affine & projective coords live here.

Reply:
generator for the curve yes, so if we call this curve the first group, then yes, g1. I'm not sure whether there exists a pairing curve for C101.